src/foam/core/stdlib.js

```diff
@@ -746,7 +747,26 @@ foam.LIB({
     methods: [
       function clone(o)      { return typeOf(o).clone(o); },
       function equals(a, b)  { return typeOf(a).equals(a, b); },
-      function compare(a, b) { return typeOf(a).compare(a, b); },
+      function compare(a, b) {
+        // To ensure that symmetry is present when comparing,
+        // we will always use the comparator of higher precedence.
+        var types = [ foam.Undefined, foam.Null, foam.Array,
```

Comment:
Let's not redefine this every time `compare` runs. Can we store it somewhere like `foam.typeOrder`? I don't love that name. @kgrgreer WDYT, re, storing this somewhere else and/or what to name it?